TypeScript: "Cannot find module 'react-dom/client'" when using React syntax with Preact + Vite

#### **Description**
When creating a new project using `create-preact` with TypeScript, using React-style imports such as:

```ts
import ReactDOM from 'react-dom/client';
```

results in this TypeScript error:

```
Cannot find module 'react-dom/client' or its corresponding type declarations.ts(2307)
```

even though the app runs fine — because `@preact/preset-vite` correctly aliases React to Preact at runtime.

---

#### **Root Cause**
The Vite preset (`@preact/preset-vite`) provides runtime aliasing for:
- `react` → `preact/compat`
- `react-dom` → `preact/compat`

However, TypeScript does not automatically inherit these aliases, so it fails to resolve subpath imports like `react-dom/client`.

---

#### **Current Workaround**
Manually add the following paths to `tsconfig.json`:

```json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "preact",
    "paths": {
      "react": ["./node_modules/preact/compat"],
      "react-dom": ["./node_modules/preact/compat"],
      "react-dom/*": ["./node_modules/preact/compat/*"] // 👈 this line fixes it
    }
  }
}
```

The critical fix is:

```json
"react-dom/*": ["./node_modules/preact/compat/*"]
```

Once this is added, TypeScript correctly resolves all React DOM submodules like `react-dom/client`.

---

#### **Suggested Fix**
Update `create-preact`’s TypeScript template so that the generated `tsconfig.json` automatically includes:

```json
"paths": {
  "react": ["./node_modules/preact/compat"],
  "react-dom": ["./node_modules/preact/compat"],
  "react-dom/*": ["./node_modules/preact/compat/*"]
}
```

This ensures smooth developer experience when using React syntax with Preact compat, without requiring any manual fixes.

---

#### **Reproduction Steps**
1. Create a new project:
   ```bash
   npm create preact@latest my-app -- --template typescript
   ```
2. Modify `main.tsx` to use React-style imports:
   ```ts
   import ReactDOM from 'react-dom/client';
   ```
3. Observe the TypeScript error:
   ```
   Cannot find module 'react-dom/client' or its corresponding type declarations.ts(2307)
   ```

---

#### **Expected Behavior**
TypeScript should recognize `react-dom/client` as `preact/compat/client` automatically  
— with no manual configuration required.

---

#### **Environment**
- **Tooling:** `create-preact` (latest)
- **Bundler:** Vite
- **Language:** TypeScript
- **Runtime:** Preact with React-style syntax
- **Error:** `Cannot find module 'react-dom/client'`

---

#### **Additional Context**
This small `paths` addition in the generated `tsconfig.json` would improve the out-of-the-box DX  
for developers using React syntax with Preact, especially since the runtime aliasing already works perfectly.

I initially wanted to edit the source and open a PR to fix this myself,  
but I wasn’t sure which files inside `create-preact` generate the TypeScript template,  
so I decided to open this issue instead to point it out 😅

--- a/templates/base/jsconfig.json
+++ b/templates/base/jsconfig.json
@@ -15,7 +15,6 @@
 			"react": ["./node_modules/preact/compat/"],
 			"react-dom": ["./node_modules/preact/compat/"],
 			"react-dom/*": ["./node_modules/preact/compat/*"]
-
 		}
 	},
 	"include": ["node_modules/vite/client.d.ts", "**/*"]

--- a/templates/base/jsconfig.json
+++ b/templates/base/jsconfig.json
@@ -13,7 +13,9 @@
 		"skipLibCheck": true,
 		"paths": {
 			"react": ["./node_modules/preact/compat/"],
-			"react-dom": ["./node_modules/preact/compat/"]
+			"react-dom": ["./node_modules/preact/compat/"],
+			"react-dom/*": ["./node_modules/preact/compat/*"]
+
 		}
 	},
 	"include": ["node_modules/vite/client.d.ts", "**/*"]

